Wetness stuff.

nullbear opened the pull request:
:cl: nullbear
tweak: Wetness updates less often.
rscadd: Wetness now stacks. Spraying lube over a tile that already has lube, will make the lube take longer to dry. Likewise, dumping 100 units of lube will take longer to dry than if only 10 units were dumped. (lube used to last just as long, regardless of whether it was 1u, or 100u) There is a limit, however, and you can't stack infinite lube.
tweak: Hotter temperatures cause water to evaporate more quickly. At 100C, water will boil away instantly.
rscadd: About 5 seconds of wetness are removed from a tile for each unit of drying agent. Absorbant Galoshes remain the most effective method of drying, and dry tiles instantly regardless of wetness.
/:cl:

nullbear commented:
-doesnt even bother to test this-

Doesnt need to update every cycle. Doesn't need to tied to linda.

optimumtact commented on code/game/turfs/open.dm:
```diff
@@ -63,7 +64,16 @@
 			if(C.m_intent=="walk" && (lube&NO_SLIP_WHEN_WALKING))
 				return 0
-
+		if(istype(C, /mob/living/carbon/human))
```
hmm, is it worth defining a mob proc process_walk and moving some of the checks into that?

This is a pretty deeply nested if chain

nullbear replied:
im pretty sure only humans have shoes. 

the next if could be reduced, though.

optimumtact replied:
yeah you're probably right

phil235 commented on code/game/turfs/open.dm:
```diff
@@ -63,7 +64,16 @@
+					if(lube&SLIDE)
+						for(var/i=1, i<4, i++)
+							spawn (i)
+								step(C, olddir)
```
This will cause the same issue as the old 'hulk not falling on lube but sliding' bug.

Make a row of slippery turf, get noslip shoes, walk on the first turf, and before the sliding stops, when you are still in the midle of the row, try to walk back. Result: you're crashing the game with lag, congratulation.

nullbear replied:
Yeah i'm fixing it right now.

nullbear replied:
-forgets to actually commit it-

nullbear commented:
Y'know, i'll just do the slippy stuff in a different PR

nullbear committed:
wetness stuff

nullbear committed:
rebalances durations.

Rebalances durations because i increased the handlewet timer.

Bawhoppen commented:
you said this fixes gluon grenades? (they should last several minutes still) if so pls merge soon

nullbear commented:
yeah im assuming the issue was with wetness updating too often. If it doesnt fix it, i'll fix it in a different pr

nullbear committed:
removes anything to do with slippy stuff

Also prevents frequent updating caused by pouring slippy stuff over an
already updating turf.

phil235 commented on code/game/turfs/open.dm:
```diff
@@ -86,7 +86,10 @@
 		return 1
 
-/turf/open/proc/MakeSlippery(wet_setting = TURF_WET_WATER) // 1 = Water, 2 = Lube, 3 = Ice
+/turf/open/proc/MakeSlippery(wet_setting = TURF_WET_WATER, min = 0, max = 0) // 1 = Water, 2 = Lube, 3 = Ice
```
the comment on this line seems a bit outdated since we use defines for wet_setting now.

Also the min and max argument needs a name change. The names are too generic and could be mistaken for ovject vars, the names should be self-explantory so as to make their function obvious. Finally the min and max arg here do not function like minimum and maximum (demonstrated by the fact that the value of min is always superior to the max in every place the proc is called).

nullbear replied:
yeah, it functions more as a min/add as opposed to min/max.

Man-Shroom- commented:
JUST MERGE IT

phil235 commented:
nullbear didn't adress my comment about min max arguments...

optimumtact commented:
#17822

Nyvrem commented:
This removes all the fun of playing Janitor, because now either you get spacecleaner or people will fuck you with stunprods/battons and throw you out an airlock. Which is especially fun when you can not get spacecleaner.

Before it already took pretty damn long for floors to stop being wet, but now it takes literal ages.

phil235 commented:
This is broken, the wetness never decreases. Somehow HandleWet() is not called (manually calling it works fine) @nullbear

nullbear commented:
bleh it worked fine when i tested it. Should be triggered whenever a surface is first made wet.
